packages/atclient/include/atclient/connection.h

```diff
@@ -14,9 +14,30 @@
 // represents the type of connection
 typedef enum atclient_connection_type {
   ATCLIENT_CONNECTION_TYPE_ATDIRECTORY, // uses '\n' to check if it is connected
-  ATCLIENT_CONNECTION_TYPE_ATSERVER // uses 'noop:0\r\n' to check if it is connected
+  ATCLIENT_CONNECTION_TYPE_ATSERVER     // uses 'noop:0\r\n' to check if it is connected
 } atclient_connection_type;
 
+typedef int(atclient_connection_send_hook)(const unsigned char *src, const size_t srclen, unsigned char *recv,
+                                           const size_t recvsize, size_t *recvlen);
+
+typedef enum atclient_connection_hook_type {
+  ACHT_NONE = 0,
+  ACHT_PRE_SEND,
+  ACHT_POST_SEND,
+  ACHT_PRE_RECV,
+  ACHT_POST_RECV,
```

Comment:
Would it be possible to expand the acronym to `ATCLIENT_CONNECTION_HOOK_TYPE_*`?

Will this enum be typed by the caller enough times to justify having this be shortened?

I think we should expand it for 3 reasons:
1. intellisense
2. less confusing on what `ACHT` means
3. guaranteed namespacing (there's a chance someone will take `ACHT_*` but no chance someone will take `ATCLIENT_CONNECTION_HOOK_TYPE*`